src/Data.Json.JSemantic.purs

```diff
@@ -0,0 +1,105 @@
+module Data.Json.JSemantic
+  ( JSemantic(..)
+  , toSemantic
+  ) where
+
+import Data.Argonaut.JCursor(JsonPrim(..), runJsonPrim)
+import Data.Maybe
+import Data.String.Regex (RegexFlags(..), regex, test)
+import Data.String
+import Data.Foldable
+import qualified Data.Array as A
+import qualified Data.Array.Unsafe as AU
+import Data.Tuple
+import qualified Data.Date as Date
+import Math
+
+
+data JSemantic = Integral   Number
+               | Fractional Number
+               | Date       Date.Date
+               | DateTime   Date.Date
+               | Time       Date.Date
+               | Interval   Date.Date Date.Date
+               | Text       String
+               | Bool       Boolean
+               | Percent    Number
+               | Currency   Number
+               | NA
+
+instance showJSemantic :: Show JSemantic where
```

Comment:
It's general convention that a `Show` instance should render a string that is code which can reconstruct value being shown (e.g. `Show (Integral n)` would be defined as `"Integral " ++ show n`). For other uses, it's best to just define a custom function (e.g. `renderJSemantic`) which has whatever string rendering you like.